x-pack/packages/kbn-elastic-assistant/impl/assistant/index.tsx

```diff
@@ -220,7 +220,7 @@ const AssistantComponent: React.FC<Props> = ({
   );
 
   useEffect(() => {
-    if (conversationsLoaded && Object.keys(conversations).length > 0) {
+    if (areConnectorsFetched && conversationsLoaded && Object.keys(conversations).length > 0) {
```

Comment:
This condition is in the useEffect below that calls `mutateAsync`. By having it in both, we avoid a race condition leading to this bug:

<img width="792" alt="error_toast_assistant" src="https://github.com/elastic/kibana/assets/915763/68b08e21-6aea-4bc1-ad5c-e4502e321a99">